### PR TITLE
feat: registry auth token into merged and push crio image

### PR DESCRIPTION
**Pull request by bsmithai**
## Describe your changes
- registry auth login from base64 auth token

## Issue ticket number
[CED-536](https://linear.app/cedana/issue/CED-536/image-registry-creds-in-crio-rootfs-checkpoint)

## Checklist before requesting a review
- [x] Have I performed a self-review?
- [ ] Have I added regression or unit tests (where it makes sense) for my changes?
- [ ] Have I updated the README if changes have resulted in it being out of date?
- [x]  Have I checked to ensure my PR only introduces the changes it's intended to? E.g. No extraneous formatting changes.
- [ ] Is this a breaking change? If yes, please describe areas affected and request reviews from developer stakeholders.

### CODE REVIEW

File: api/server.go [hunk 1]
```diff
@@ -284,9 +285,21 @@ func loggingStreamInterceptor(logger *zerolog.Logger) grpc.StreamServerIntercept
 	}
 }
 
+func RedactAuthToken(req interface{}) interface{} {
+	val := reflect.ValueOf(req).Elem()
+	for i := 0; i < val.NumField(); i++ {
+		field := val.Field(i)
+		if field.Kind() == reflect.String && (val.Type().Field(i).Name == "RegistryAuthToken") {
```
Comment (nravic) [inline api/server.go:292]: This is a great idea, can you also add redactions to env vars? so if Name contains the phrase KEY or something we redact

**Review by nravic — APPROVED** [review]
lgtm, small comment

File: api/server.go (hunk 1 above)
Comment (bsmithai) [reply api/server.go:292]: yessir

**Review by nravic — APPROVED** [review]
lgtm

**Pull request merged**